src/main/java/com/minecolonies/coremod/util/InventoryUtils.java

```diff
@@ -346,54 +346,59 @@ public static boolean isPickaxeInItemHandler(IItemHandler itemHandler, final int
-                                                                        && InventoryUtils.hasToolLevel(stack, Utils.PICKAXE, maximalLevel));
+                                                                        && InventoryUtils.hasToolLevel(stack, Utils.PICKAXE, requiredLevel, maximalLevel));
     }
 
     /**
      * Verifies if there is one tool with an acceptable level
      * in a worker's inventory.
      *
-     * @param toolName     the type of tool needed
      * @param stack        the stack to test.
-     * @param minimalLevel the worker's hut level
+     * @param toolName     the type of tool needed
+     * @param minimalLevel the minimum level for the tool to find.
+     * @param maximumLevel the maximum level for the tool to find.
      * @return true if tool is acceptable
      */
-    public static boolean hasToolLevel(@Nullable ItemStack stack, final String toolName, final int minimalLevel)
+    public static boolean hasToolLevel(@Nullable ItemStack stack, final String toolName, final int minimalLevel, final int maximumLevel)
     {
         if (isItemStackEmpty(stack))
         {
             return false;
         }
 
         final int level = Utils.getMiningLevel(stack, toolName);
-        if (Utils.isTool(stack, toolName) && verifyToolLevel(stack, level, minimalLevel))
-        {
-            return true;
-        }
-
-        return false;
+        return Utils.isTool(stack, toolName) && verifyToolLevel(stack, level, minimalLevel, maximumLevel);
     }
 
     /**
      * Verifies if an item has an appropriated grade.
      *
      * @param itemStack    the type of tool needed
-     * @param toolLevel    the type of tool needed
-     * @param minimalLevel the worker's hut level
+     * @param toolLevel    the tool level
+     * @param minimalLevel the minimum level needed
+     * @param maximumLevel the maximum level needed (usually the worker's hut level)
      * @return true if tool is acceptable
      */
-    public static boolean verifyToolLevel(@NotNull final ItemStack itemStack, int toolLevel, final int minimalLevel)
+    public static boolean verifyToolLevel(@NotNull final ItemStack itemStack, int toolLevel, final int minimalLevel, final int maximumLevel)
     {
-        if (isItemStackEmpty(itemStack) || minimalLevel > FREE_TOOL_CHOICE_LEVEL || minimalLevel >= toolLevel)
+        if (isItemStackEmpty(itemStack))
         {
+            //We can always use a bare hand
             return true;
         }
-        else if (itemStack.isItemEnchanted() && minimalLevel <= EFFECT_TOOL_CHOICE_LEVEL)
+
+        if (toolLevel < minimalLevel || toolLevel > maximumLevel)
         {
             return false;
         }
 
-        return false;
+        //Check that we can use a enchanted item
+        if (itemStack.isItemEnchanted() && maximumLevel < EFFECT_TOOL_CHOICE_LEVEL)
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this if-then-else statement by a single return statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1126)